Waidhoferj/CSC-308-309-Project
# GraphQL Integration between frontend and backend

The frontend can now pull data from the backend.

## Changelog:
- Added option to launch a local server instead of using the cloud instance by calling `python app.py -l`
- Dynamically imports `secrets.py`. If user doesn't have it, launches the mock instance instead of erroring.
- Made some updates to the mock server.
- Added CORS to the flask server so that the frontend can access the api
- Added Apollo Client to the frontend in order to cache fetches from graphql api.
- Integrated the map with the graphql api. Data is now pulled directly from the server launched in the backend folder.
## Testing:
1. Launch the backend server in `/backend`: `pipenv install && pipenv shell && python app.py -l`
2. Launch the website in `/frontend`: `npm install && npm start`
3. Go to `/map` page.
4. Ensure that you can interact with the artworks. The data for these artworks can be found in `database.py`
## Resources:
[Apollo Client](https://www.apollographql.com/docs/react/)

Fixes #22, fixes #29, fixes #36
